app/include/zmk/event_manager.h

```diff
@@ -68,17 +71,22 @@ struct zmk_event_subscription {
             .listener = &zmk_listener_##mod,                                                       \
     };
 
-#define ZMK_EVENT_RAISE(ev) zmk_event_manager_raise((zmk_event_t *)ev);
+#define ZMK_EVENT_RAISE(ev)                                                                        \
+    ((__ASSERT((uint8_t *)&(ev).header - (uint8_t *)&ev == 0,                                      \
+               "header must be first element of event")),                                          \
+     zmk_event_manager_raise(&(ev).header));
```

Comment:
```suggestion
     zmk_event_manager_raise(&(ev).header))
```
If these should be usable in expressions like `return ZMK_EVENT_RAISE(ev)`, then they shouldn't end in a semicolon. That will prevent other things that should work like
```c
result = ZMK_EVENT_RAISE(ev) ? OK : ERROR;
```
and it also allows 
```c
int test(void) {
  return ZMK_EVENT_RAISE(ev)  // no semicolon
}
```
to compile, which just feels weird.